test/modules/generic-mqtt-tester/src/tester.rs

```diff
@@ -127,39 +134,47 @@ impl MessageTester {
         let test_result_coordinator_url = settings.trc_url();
         let reporting_client = TrcClient::new(test_result_coordinator_url);
 
-        let message_handler: Box<dyn MessageHandler + Send> = match settings.test_scenario() {
-            TestScenario::Initiate => {
-                let batch_id = settings
-                    .batch_id()
-                    .ok_or(MessageTesterError::MissingBatchId)?;
-                Box::new(ReportResultMessageHandler::new(
+        let message_handler: Option<Box<dyn MessageHandler + Send>> = match settings.test_scenario()
+        {
+            TestScenario::InitiateAndReceiveRelayed | TestScenario::Receive => {
+                Some(Box::new(ReportResultMessageHandler::new(
                     reporting_client.clone(),
                     tracking_id,
-                    batch_id,
                     &module_name,
-                ))
+                )))
             }
-            TestScenario::Relay => Box::new(RelayingMessageHandler::new(
+            TestScenario::Relay => Some(Box::new(RelayingMessageHandler::new(
                 publish_handle.clone(),
                 relay_topic,
-            )),
+            ))),
+            TestScenario::Initiate => None,
         };
```

Comment:
nit: this might be a nice helper fn to reduce fluff in the new, your call though :)

Reply:
Good feedback! I can refactor this the next time I touch the test module. Since this is a cherry pick I don't think I should make changes to this PR.